specs/agents/tracing-spans.md

```diff
@@ -17,6 +17,7 @@ for simpler and more performant UI queries.
 Each transaction has a `type` field, each span has both `type` and `subtype` fields.
 The values for each of those fields is protocol-specific and defined in the respective instrumentation specification
 for each protocol.
+If no `transaction.type` or `span.type` is provided or the value is an empty string, the agent needs to set a default value `unknown`.
```

Comment:
Set to `unknown` according to suggestions on the issue https://github.com/elastic/apm-server/issues/7376#issuecomment-1047921836.

Reply:
Might be relevant to add the `unknown` value for span `type` in the JSON spec as well : https://github.com/elastic/apm/blob/main/tests/agents/json-specs/span_types.json#L60

Currently we are using `custom` by default if it's not set (at least for the Java agent), but we don´t  enforce anything when it's set but empty.

Reply:
From what I know (maybe @felixbarny can confirm this), the meaning of `custom` is only to provide a value when not explicitly set when using the API. Thus, using `unknown` when it's set to an empty value could complement this, so we have to decide between:
- use `custom` when not set, and `unknown` when set to an empty value.
- use only one of `custom` or `unknown` when span or transaction type is **not set or empty**, I would have a slight preference for `custom` as it feels closer to using agent API.

Reply:
I guess the idea with `custom` comes from the assumption that, if it's not set then a user called the API and therefore it's a "custom" transaction/span created by a user via API - right? 

But is it safe to assume that if from whatever reason the type ends up being an empty string (this is already an edge case) then its type is `custom`? I feel that could be a bit misleading - I feel that in those case we simply don't know the type, therefore it's `unknown`, so I'd not assume `custom`.

Reply:
On the other hand, the idea with `custom` for transactions/spans via the API sound very reasonable - so basically @SylvainJuge's 1. point.

It'd be something like this:

```
If a transaction/span is created via the public agent API then the default `transaction.type` or `span.type` SHOULD be `custom`.

If the `transaction.type` or `span.type` value is an empty string or `null`, the agent SHOULD set a default value `unknown`.
```

I used "SHOULD" for both because 1) I don't think most agents currently use `custom` (we don't for .NET) in case of public API and 2) I assume if the agent does not fix empty strings, the server will. But we can flip this part to "MUST" as well.

Reply:
I think we should keep it simple and use the same default value regardless of whether the type is not set or set to an empty string. I have a slight preference for `custom` over `unknown`.
If it's feasible for all agents to change the current behavior, I think we should enforce consistency by using MUST instead of SHOULD. For agents that don't comply with that, we can create implementation issues to change the default value. I wouldn't consider that to be a breaking change.

Reply:
Ok to me. If we want to have only 1 default, then I agree `custom` is a better one.

I still think always setting to `custom` may cause problems, because there could be cases when it may not be a custom transaction/span (e.g. there is a bug and auto instrumentation does not set it (I think it's better to end up with `unknown` than `custom`), or some other thing) and if a user forces an empty string I'd still handle it differently.

But this is an edge case anyways, not worth blocking this - if people want to go with only `custom` and keep it simple, it's fine by me.

Reply:
```suggestion
If no `transaction.type` or `span.type` is provided or the value is an empty string, the agent needs to set a default value `custom`.
```